Project-Hotpot/Project-Hotpot/Parse-Backend/PFPlaylist.swift

```diff
@@ -33,34 +34,66 @@ class PFPlaylist: PFObject, PFSubclassing {
         })
     }
     
-    class func addSongToPlaylistInBackground(song: PFSong,
-                                             playlist: PFPlaylist,
-                                             completion: @escaping((Bool, Error?) -> Void)) {
-        let relation = playlist.relation(forKey: "songs")
-        relation.add(song)
-        playlist.saveInBackground {success, error in
-            if success == true {
+    
+    class func addSongtoPlaylistInBackground(song: PFSong, playlist: PFPlaylist, completion: @escaping ((Bool, Error?)-> Void)){
+        // create an entry in the Follow table
+        let joinTable = PFObject(className: "SongJoinTable")
+        joinTable.setObject(song, forKey: "Song")
+        joinTable.setObject(playlist, forKey: "Playlist")
+        
+        let currentDate = NSDate()
+        joinTable.setObject(currentDate, forKey: "AddedAt")
+        joinTable.setObject(currentDate, forKey: "LastPlayed")
+        
+        joinTable.saveInBackground { saved, error in
+            if saved {
+                NSLog("song saved successfully")
                 completion(true, nil)
             }
-            else{
+            else {
+                NSLog("failed to save song to playlist")
                 completion(false, error)
             }
         }
     }
     
-    //TODO: change to getting ALL playlists instead of just first.
-    class func getPlaylistInBackground(completion: @escaping(PFPlaylist?, Error?) -> Void) {
+    class func getAllSongsFromPlaylist(playlist: PFPlaylist, completion: @escaping ([PFSong]?, Error?)->Void){
+        let query = PFQuery(className: "SongJoinTable")
+        var songArray: [PFSong]? = []
```

Comment:
Nit: bring this declaration closer to where it's actually used.
Also, is there a need to initialize it to an empty array since it's optional? It feels like this should either be initialized to nil, or made non-optional.